src/mars_patcher/zm/item_patcher.py

```diff
@@ -0,0 +1,77 @@
+from mars_patcher.rom import Rom
+from mars_patcher.room_entry import RoomEntry
+from mars_patcher.zm.auto_generated_types import MarsschemazmTankIncrements
+from mars_patcher.zm.constants.game_data import (
+    chozo_statue_targets_addr,
+    major_locations_addr,
+    minor_locations_addr,
+    tank_increase_amounts_addr,
+)
+from mars_patcher.zm.locations import HintLocation, LocationSettings
+
+
+class ItemPatcher:
+    """Class for writing item assignments to a ROM."""
+
+    def __init__(self, rom: Rom, settings: LocationSettings):
+        self.rom = rom
+        self.settings = settings
+
+    def write_items(self) -> None:
+        rom = self.rom
+        hint_targets_addr = chozo_statue_targets_addr(rom)
+
+        # Handle minor locations
+        # Locations need to be written in order so that binary search works
+        minor_locs = sorted(self.settings.minor_locs, key=lambda x: x.key)
+        minor_loc_addr = minor_locations_addr(rom)
+        for min_loc in minor_locs:
+            # TODO: Update tileset and get BG1 value
+            bg1_val = 0x4A  # Use power bomb tank block for now
+            # Overwrite BG1 if not hidden
+            if not min_loc.hidden:
+                room = RoomEntry(rom, min_loc.area, min_loc.room)
+                with room.load_bg1() as bg1:
+                    bg1.set_block_value(min_loc.block_x, min_loc.block_y, bg1_val)
+
+            # See struct MinorLocation in include/structs/randomizer.h
+            rom.write_32(minor_loc_addr, min_loc.key)
+            rom.write_16(minor_loc_addr + 4, bg1_val)
+            rom.write_8(minor_loc_addr + 6, min_loc.new_item.value)
+            rom.write_8(minor_loc_addr + 7, min_loc.item_jingle.value)
+            # TODO: Handle custom messages
+            rom.write_32(minor_loc_addr + 8, 0)
+            rom.write_8(minor_loc_addr + 0xC, min_loc.hint_value)
+            minor_loc_addr += 0x10
+
+            if min_loc.hinted_by != HintLocation.NONE:
```

Comment:
i feel like this hint system is *way* too overpowered, but that's a thing for later.